### PR TITLE
Remaining subgroup validation tests

### PR DESCRIPTION
**Pull request by alan-baker**
* Adds must_use tests to ballot and broadcast
* Adds const id requirement to broadcast
* Adds tests for other builtins:
  * broadcast first,
  * elect,
  * min, max,
  * and, or, xor,
  * shuffle, shuffle xor, shuffle up, shuffle down
  * quad broadcast
  * quad swap x, quad swap y, quad swap diagonal

Issue: #

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by dneto0 — CHANGES_REQUESTED** [review]
missing must_use tests for broadcast

File: src/webgpu/shader/validation/expression/call/builtin/subgroupBroadcastFirst.spec.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
+export const description = `
```
Comment (dneto0) [inline src/webgpu/shader/validation/expression/call/builtin/subgroupBroadcastFirst.spec.ts:1]: coder's choice, but could be folded in with the subgroupBroadcast test file?

**Review by dneto0 — CHANGES_REQUESTED** [review]
Finished reviewing.

I noticed an opportunity to to update the proposal to require validation of the range of a const subgroup IDs or const mask or const ID-delta.
Posted https://github.com/gpuweb/gpuweb/issues/4306#issuecomment-2313412561 for discussion later.

File: src/webgpu/shader/validation/expression/call/builtin/subgroupBroadcastFirst.spec.ts (hunk 1 above)
Comment (alan-baker) [reply src/webgpu/shader/validation/expression/call/builtin/subgroupBroadcastFirst.spec.ts:1]: Thought about that, but they take a different number of arguments.

**Comment by dneto0** [comment]
> * ballot and broadcast

Never mind.  This test already exists in`main`.

File: src/webgpu/shader/validation/expression/call/builtin/quadBroadcast.spec.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
+export const description = `
+Validation tests for quadBroadcast
+`;
```
[the remark this answers is not in the document]
Comment (alan-baker) [reply src/webgpu/shader/validation/expression/call/builtin/quadBroadcast.spec.ts:3]: Added for all functions. For those that can use f16, added a variant to also confirm that enable.

**Review by dneto0 — APPROVED** [review]
thanks!

**Pull request merged**